server/app/services/applicant/RepeatedEntity.java

```diff
@@ -92,12 +113,23 @@ public int depth() {
     return 1 + parent().map(RepeatedEntity::depth).orElse(0);
   }
 
+  /** Gets visibility for this and all parents of the repeated entity. */
+  public ImmutableList<PredicateDefinition> nestedVisibility() {
+    return Stream.concat(
+            parent()
+                .map(p -> p.nestedVisibility())
```

Comment:
suggestion (non-blocking): the type would be more clear here with method reference i.e. `.map(PredicateDefinition::nestedVisibility)`

Reply:
done